### PR TITLE
rework sass using bem and outside-in

### PR DESCRIPTION
**Pull request by hsujessica**

### CODE REVIEW

File: src/components/GridSquare.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React, {Component} from 'react';
+import {connect} from 'react-redux';
+
+
+export class GridSquare extends Component {
+  render() {
+    return (
+      <div className="grid-square">
+        <form id="form-1">
```
Comment (hsubox) [inline src/components/GridSquare.jsx]: how is each square it's own form?

File: src/components/LoginForm.jsx [hunk 2]
```diff
@@ -19,9 +19,9 @@ class LoginForm extends Component {
     } = this.props;
     return (
       <div className='login-form'>
-        <form>
-          <InputTextField fieldName='Email' placeholder='Email' value={email} onChange={loginFormUpdate}/>
-          <InputTextField type='password' placeholder='Password' fieldName='Password' value={password} onChange={loginFormUpdate}/>
+        <form className='login-form__form'>
```
Comment (hsubox) [inline src/components/LoginForm.jsx]: not everything has to have __. this could just be login-form

File: src/components/MainNavBar.jsx [hunk 3]
```diff
@@ -8,7 +8,7 @@ class MainNavBar extends Component {
     return (
       <div className='main-nav-bar'>
         <nav>
-          <Link to='/'>MENU +</Link>
+          <Link to='/' className='main-nav-bar__a'>MENU +</Link>
```
Comment (hsubox) [inline src/components/MainNavBar.jsx]: what is __a? maybe a __link?

File: src/styles/components/grid-square.scss [hunk 4]
```diff
@@ -0,0 +1,29 @@
+.grid-square {
+  display: inline-flex;
+  flex-direction: row;
+  margin: 2rem;
+}
+
+.grid-square__input[type="checkbox"] {
```
Comment (hsubox) [inline src/styles/components/grid-square.scss]: why do we need the checkbox type? did you consider a --checkbox if you really need to check that specifically?

File: src/styles/components/grid-square.scss [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  position: relative;
+  margin: auto;
+  padding: 10px 18px;
+  height: 1px;
+  width: 1px;
+  background-color: #E7E7E7;
+  border: 1px dotted #DFF8EB;
+}
+
+.grid-square__input[type="checkbox"]:checked + label {
```
Comment (hsubox) [inline src/styles/components/grid-square.scss]: rework this

File: src/styles/components/grid-square.scss [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+.grid-square__input[type="checkbox"] {
+  display: flex;
+  flex-direction: column;
+  margin: .65em;
+  height: 1px;
+  width: 1px;
+  opacity: 0;
+}
+
+.grid-square__label {
```
Comment (hsubox) [inline src/styles/components/grid-square.scss]: we are in the SASS world? you should be able to nest these properties. e.g.

```
.grid-square {
  ... stuff
  &__label {
       .. stuff
   }
}
```

File: src/styles/components/header.scss [hunk 7]
```diff
@@ -3,8 +3,9 @@
...
   align-items: center;
   background-color: #DFF8EB;
-  img {
-    height: 2rem;
-    padding: 2rem;
-  }
+}
+
+.header__image {
+  height: 2rem;
```
Comment (hsubox) [inline src/styles/components/header.scss]: choose a different unit please

File: src/styles/components/login-form.scss [hunk 8]
```diff
@@ -1,24 +1,15 @@
...
-    background-color: #DFF8EB;
-  }
-  a:hover, a:active, a:visited {
-    background-color: rgba(242,77,85,.7);
-  }
+.login-form__form {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  margin: 3em auto;
```
Comment (hsubox) [inline src/styles/components/login-form.scss]: no ems

**Review by hsubox — CHANGES_REQUESTED** [review]
please make the suggested edits

File: src/components/LoginForm.jsx (hunk 2 above)
Comment (hsujessica) [reply src/components/LoginForm.jsx]: the div it is enclosed in also has the class login-form...?

Comment (hsubox) [reply src/components/LoginForm.jsx]: how about __container then?

File: src/styles/components/grid-square.scss (hunk 4 above)
Comment (hsujessica) [reply src/styles/components/grid-square.scss]: can you explain? i don't understand your question

File: src/styles/components/grid-square.scss (hunk 5 above)
Comment (hsubox) [reply src/styles/components/grid-square.scss]: [type="checkbox"] -> part of the class name
label -> class name
no +'s

File: src/styles/components/grid-square.scss (hunk 4 above)
Comment (hsubox) [reply src/styles/components/grid-square.scss]: see reply below

**Review by hsubox — APPROVED** [review]

**Pull request merged**